src/main/java/com/karankumar/bookproject/backend/utils/CsvUtils.java

```diff
@@ -0,0 +1,31 @@
+package com.karankumar.bookproject.backend.utils;
+
+import com.fasterxml.jackson.databind.MappingIterator;
+import com.fasterxml.jackson.dataformat.csv.CsvMapper;
+import com.fasterxml.jackson.dataformat.csv.CsvSchema;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+
+@Slf4j
+public final class CsvUtils {
+    public static final String TEXT_CSV = "text/csv";
+    private static final CsvMapper csvMapper;
+
+    static {
+        csvMapper = new CsvMapper();
+        csvMapper.registerModule(new JavaTimeModule());
+    }
+
+    private CsvUtils() {}
+
+    public static <T> List<T> read(InputStream inputStream, Class<T> classType) throws IOException {
```

Comment:
What would happen if the file type is not csv? Do we want to fail fast with some error message about the file type?

Reply:
I'm not entirely sure if we can add a validation here for file type, as we are using InputStream here and the underlying implementation may not necessarily be a FileInputStream.